lib/openstudio_measure_tester/github_actions_report.rb

```diff
@@ -0,0 +1,110 @@
+# frozen_string_literal: true
+
+# *******************************************************************************
+# OpenStudio(R), Copyright (c) Alliance for Sustainable Energy, LLC.
+# See also https://openstudio.net/license
+# *******************************************************************************
+
+module OpenStudioMeasureTester
+  class GithubActionsReport
+    attr_reader :minitest_summary_table, :all_annotations
+
+    # @param test_results_directory [String]: The directory
+    def initialize(test_results_directory)
+      @test_results_directory = test_results_directory
+      file = File.read("#{@test_results_directory}/combined_results.json")
+      @hash = JSON.parse(file)
+      @minitest_summary_table = nil
+      @all_annotations = []
+    end
+
+    def write_step_summary(msg)
+      puts msg
+      if !ENV['GITHUB_STEP_SUMMARY'].nil?
+        File.write(ENV['GITHUB_STEP_SUMMARY'], "#{msg}\n", mode: 'a+')
+      end
+    end
+
+    def hash_to_markdown(h, header0, header1)
+      n0 = [h.keys.map(&:to_s).map(&:size).max, header0.size].max
+      n1 = [h.values.map(&:to_s).map(&:size).max, header1.size].max
+
+      content = [
+        "| #{header0.ljust(n0)} | #{header1.ljust(n1)} |",
+        "| " + "-" * n0 + " | " + "-" * n1 + " |",
+      ] + h.map{|k, v| "| #{k.ljust(n0)} | #{v.to_s.ljust(n1)} |"}
+      return content.join("\n")
+    end
```

Comment:
Just a cleverish way to format a markdown table so it looks good even if printed locally to your console. I didn't want to add a dependency on another gem.

```
| Metric           | Value  |
| ---------------- | ------ |
| Total Tests      | 16     |
| Load Error       | 0      |
| Passed           | 13     |
| Success Rate     | 86.67% |
| Failures         | 1      |
| Errors           | 1      |
| Skipped          | 1      |
| Incompatible     | 1      |
| Total Assertions | 120    |
```

| Metric           | Value  |
| ---------------- | ------ |
| Total Tests      | 16     |
| Load Error       | 0      |
| Passed           | 13     |
| Success Rate     | 86.67% |
| Failures         | 1      |
| Errors           | 1      |
| Skipped          | 1      |
| Incompatible     | 1      |
| Total Assertions | 120    |